app/connectors/HmrcTierConnector.scala

```diff
@@ -40,33 +38,71 @@ class HmrcTierConnector @Inject() (client: HttpClient)(implicit ec: ExecutionCon
     }
   }
 
-  def genericGetCall[T](baseUrl: String, URIExtension: String, empRef: EmpRef, year: Int)(implicit
-    hc: HeaderCarrier,
-    formats: Format[T]
-  ): Future[T] = {
-    val resp = client.GET(createGetUrl(baseUrl, URIExtension, empRef, year))
+  def get(baseUrl: String, URIExtension: String, empRef: EmpRef, year: Int)(implicit hc: HeaderCarrier): Future[HttpResponse] = {
+
+    val request = client.GET(createGetUrl(baseUrl, URIExtension, empRef, year))
+    request
+  }
+
+  /*
+    - New code should be simple and not do too many things
+    - We can potentially still keep the logging of headers and validation
+    - However, returning the response as is opens up the possibility to transform the data into whatever we want.
+    - Old genericGet method was returning a tuple making it annoying to manipulate as you would have to deconstruct the tuple into ._1 or ._2 etc.
```

Comment:
sorry, referring to methods at the service level,  they were tupling up headers from the var Map,
when genericGet was being called to get List[Bik]